src/Symfony/Component/HtmlSanitizer/Tests/Parser/MastermindsParserTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer\Tests\Parser;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HtmlSanitizer\Parser\MastermindsParser;
+
+class MastermindsParserTest extends TestCase
+{
+    public function testParseValid()
```

Comment:
The parser never throws exception (which is why I removed the exception mechanism I had originally). It will always return a string, and if there have been parsing errors it will provide them in an independent method call. If one wants to get these errors themselves, they could use a customer ParserInterface.

Testing for invalid HTML is therefore already done in the HtmlSanitizer tests, where invalid HTML tags are provided to ensure no unsafe output is generated.